app/controllers/concerns/course/users_staff_statistics_concern.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+module Course::UsersStaffStatisticsConcern
+  extend ActiveSupport::Concern
+
+  # Sort the staff by their average marking time.
+  # Note that nil time will be considered as the largest, which will come to the bottom of the
+  #   list.
+  #
+  # @return [Array<CourseUser>] CourseUsers, sorted by average marking time.
+  def order_by_average_marking_time(staff)
```

Comment:
👍  seems like a good suggestion, although this should probably be a method within the association >_<